src/amo/reducers/featured.js

```diff
@@ -0,0 +1,33 @@
+import {
+  FEATURED_GET,
+  FEATURED_LOADED,
+  FEATURED_FAILED,
+} from 'core/constants';
+
+const initialState = {
+  count: 0,
+  loading: false,
+  results: [],
```

Comment:
I see three separate reducers: `featured`, `highlyRated`, and `popular` but they all look nearly identical. I also see a lot of repetition in all the code and tests that cover their logic. What if you combined them all into one reducer and just tracked the "buckets" of add-on results in separate keys? more like:

````javascript
const initialState = {
  featured: {
    count: 0,
    loading: false,
    results: [],
  },
  highlyRated: {
    count: 0,
    loading: false,
    results: [],
  },
  popular: {
    count: 0,
    loading: false,
    results: [],
  },
};
````